interfaces/cython/cantera/examples/reactors/reactor3.py

```diff
@@ -0,0 +1,61 @@
+"""
+Constant-pressure, adiabatic kinetics simulation.
+Simulation is equivalent to reactor1.py, except for an automatically 
+reduced step size when large changes of temperature are anticipated.
+"""
+
+import sys
+import numpy as np
+
+import cantera as ct
+
+gas = ct.Solution('gri30.xml')
+gas.TPX = 1001.0, ct.one_atm, 'H2:2,O2:1,N2:4'
+r = ct.IdealGasConstPressureReactor(gas)
+
+sim = ct.ReactorNet([r])
+time = 0.0
+states = ct.SolutionArray(gas, extra=['t'])
+
+# limit advance when temperature difference is exceeded
+ix = r.component_index('temperature')
+delta_T_max = 20.
+sim.set_atol_advance(ix, delta_T_max)
+sim.verbose = True
+
+print('%10s %10s %10s %14s' % ('t [s]','T [K]','P [Pa]','u [J/kg]'))
+for n in range(100):
+    time += 1.e-5
+    while sim.advance_towards(time) < time:
+        states.append(r.thermo.state, t=sim.time*1e3)
+        print('%10.3e %10.3f %10.3f %14.6e' % (sim.time, r.T,
+                                               r.thermo.P, r.thermo.u))
+    states.append(r.thermo.state, t=time*1e3)
+    print('%10.3e %10.3f %10.3f %14.6e' % (sim.time, r.T,
+                                           r.thermo.P, r.thermo.u))
```

Comment:
I would strongly prefer not to use the old-style `%` formatting. Since Python 3.5 is retired from Red Hat support this month (!), we should consider using the 3.6-and-up "f"-strings. If not that, at least the `str.format` method should be used.

Reply:
> I would strongly prefer not to use the old-style `%` formatting. 

Noted. Will convert when copying code going forward

Reply:
Python 3.5 isn't reaching end-of-life from python.org until [September 2020](https://devguide.python.org/#status-of-python-branches), so I think it's a little premature to make 3.6 the minimum requirement already.

Reply:
> You can avoid the nested loop and achieve essentially the same effect by writing this as:
> 
> ```python
> while sim.time < 1e-3:
>     sim.advance_towards(sim.time + 1e-5)
>     states.append(r.thermo.state, t=sim.time*1e3)
>     print('%10.3e %10.3f %10.3f %14.6e' % (sim.time, r.T,
>                                            r.thermo.P, r.thermo.u))
> ```

It doesn't do quite the same: my intent for `advance_towards` was to generate output with prescribed step size when changes are moderate, and add interstitial points when needed, i.e. something that is more similar to `advance` than to `step`. 

As `sim.time` is moved forward by `advance_towards`,  you will end up with different output times, as the 1e-5 will always be added to the last time. For this example, I wanted to get the same output points as the regular `advance` with added points when needed.

Reply:
@speth Well, that's surprising. I assumed RHEL would be the last to support a given version of Python. Then `str.format` should be preferred, I suppose.

Reply:
Switched to `string.format`.